src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,80 @@
-import React from 'react';
+import { useCallback, useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { setUserFromApiById } from '../../utils/setUserFromApiById';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { currentTodoSlice } from '../../features/currentTodo';
+
+export const TodoModal = () => {
+  const [user, setUser] = useState<User>();
+  const [isLoading, setIsLoading] = useState(true);
+
+  const selectedTodo = useAppSelector(state => state.currentTodo);
+  const selectedTodoDispatch = useAppDispatch();
+
+  const changeCurrentTodo = useCallback(
+    (newTodos: Todo | null) => {
+      selectedTodoDispatch(
+        currentTodoSlice.actions.changecurrentTodo(newTodos),
+      );
+    },
+    [selectedTodoDispatch],
+  );
+
+  useEffect(() => {
+    if (selectedTodo !== null) {
+      setUserFromApiById(selectedTodo.userId, setUser, setIsLoading);
+    }
+  }, [selectedTodo]);
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
+      {isLoading ? (
+        <Loader />
+      ) : (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              Todo #{selectedTodo?.id || 0}
```

Comment:
The fallback value `0` for `selectedTodo?.id` might not be appropriate if `0` is a valid ID. Consider using a more distinct value or handling the null case differently.